src/components/multiSelect/MultiSelectBottomSheet.tsx

```diff
@@ -0,0 +1,232 @@
+import GorhomBottomSheet, { BottomSheetProps } from '@gorhom/bottom-sheet'
+import React, { useRef } from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import FastImage from 'react-native-fast-image'
+import { ScrollView } from 'react-native-gesture-handler'
+import BottomSheetBase from 'src/components/BottomSheetBase'
+import BottomSheetScrollView from 'src/components/BottomSheetScrollView'
+import Touchable from 'src/components/Touchable'
+import Checkmark from 'src/icons/Checkmark'
+import Colors from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+
+const OPTION_HEIGHT = 60
+const MAX_OPTIONS_IN_VIEW = 5
+
+interface MultiSelectBottomSheetProps<T extends string> {
+  forwardedRef: React.RefObject<GorhomBottomSheet>
+  onChange?: BottomSheetProps['onChange']
+  onClose?: () => void
+  onOpen?: () => void
+  handleComponent?: BottomSheetProps['handleComponent']
+  options: Option[]
+  selectedOptions: Option[]
+  setSelectedOptions: (selectedOptions: Option[]) => void
+  selectAllText: string
+  title: string
+}
+
+export interface Option {
+  id: string
+  text: string
+  iconUrl?: string
+}
+
+function MultiSelectBottomSheet<T extends string>({
+  forwardedRef,
+  onClose,
+  onOpen,
+  options,
+  setSelectedOptions,
+  selectedOptions,
+  selectAllText,
+  title,
+}: MultiSelectBottomSheetProps<T>) {
+  const { t } = useTranslation()
+  const scrollViewRef = useRef<ScrollView>(null)
+
+  const isEveryOptionSelected = options.length === selectedOptions.length
+
+  const handleClose = () => {
+    onClose?.()
+  }
+
+  return (
+    <BottomSheetBase
+      forwardedRef={forwardedRef}
+      onClose={handleClose}
+      onOpen={onOpen}
+      backgroundStyle={{ backgroundColor: 'transparent' }}
+      handleIndicatorStyle={{ width: 0 }}
+    >
+      <BottomSheetScrollView
+        forwardedRef={scrollViewRef}
+        testId={'MultiSelectBottomSheet'}
+        containerStyle={styles.bottomSheetScrollView}
+      >
+        <View style={[styles.option, styles.borderRadiusTop]}>
+          <Text style={styles.boldTextStyle}>{title}</Text>
+        </View>
+        <View style={styles.optionsContainer}>
+          <ScrollView style={{ height: OPTION_HEIGHT * MAX_OPTIONS_IN_VIEW }}>
+            <Option
+              text={selectAllText}
+              isSelected={isEveryOptionSelected}
+              onPress={() => setSelectedOptions(options)}
+            />
+            {options.map((option) => (
+              <Option
+                text={option.text}
+                iconUrl={option.iconUrl}
+                isSelected={
```

Comment:
This all select behavior differs from the [material design standards](https://m3.material.io/components/checkbox/accessibility) on parent check boxes. Also left a comment in [Figma](https://www.figma.com/file/gFuRy5HJrV34ehkY0dvWWb?type=design&node-id=2632-4282&mode=design#745860228).

Reply:
waiting on nitya's response. I'd prefer if it didn't block the pr :)